common/buildcraft/builders/tile/TileQuarry.java

```diff
@@ -325,17 +331,46 @@ public void invalidate() {
         BCBuildersEventDist.INSTANCE.invalidateQuarry(this);
         if (!world.isRemote) {
             world.removeEventListener(worldEventListener);
+            ChunkLoaderManager.releaseChunksFor(this);
         }
     }
 
+    @Nullable
     @Override
-    public void onLoad() {
-        frameBoxPoses.clear();
-        if (frameBox.isInitialized()) {
-            frameBoxPoses.addAll(frameBox.getBlocksInArea());
+    public LoadType getLoadType() {
+        return LoadType.HARD;
+    }
+
+    @Nullable
+    @Override
+    public Collection<ChunkPos> getChunksToLoad() {
+        if (!miningBox.isInitialized()) {
+            return null;
+        }
+        ArrayList<ChunkPos> list = new ArrayList<>();
+        int minX = miningBox.min().getX() >> 4;
+        int minZ = miningBox.min().getZ() >> 4;
+        int maxX = miningBox.max().getX() >> 4;
+        int maxZ = miningBox.max().getZ() >> 4;
+        for (int x = minX; x < maxX; x++) {
+            for (int z = minZ; z < maxZ; z++) {
+                list.add(new ChunkPos(x, z));
+            }
+        }
+        return list;
+    }
+
+    private void loadFrameCaches() {
+        IBlockState state = world.getBlockState(pos);
+        if (state.getBlock() != BCBuildersBlocks.quarry) {
+            frameLocations = Collections.emptyList();
         }
         toCheck.clear();
-        toCheck.addAll(frameBoxPoses);
+        if (frameBox.isInitialized()) {
+            toCheck.addAll(frameBox.getBlocksInArea());
```

Comment:
but we don't want the edges, we want to check not only the frame locations but the entire box encompased by the frames has to be empty for the drill to have room to work